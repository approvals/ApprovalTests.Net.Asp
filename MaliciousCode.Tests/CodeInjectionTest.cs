using System;
using System.Web.Mvc;
using ApprovalTests;
using ApprovalTests.Asp;
using ApprovalTests.Asp.Mvc;
using ApprovalTests.Reporters;
using ApprovalUtilities.Utilities;
using CassiniDev;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using MvcApplication1;
using MvcApplication1.Controllers;
using MvcApplication1.Models;
using ApprovalTests.Scrubber;
using ApprovalUtilities.Utilities;
using ApprovalTests.Asp.Mvc.Utilities;
using System.Text.RegularExpressions;

namespace MaliciousCode.Tests
{
    [TestClass]
    [UseReporter(typeof(DiffReporter))]
    public class CodeInjectionTest
    {
        private readonly CassiniDevServer server = new CassiniDevServer();

        [TestInitialize]
        public void Setup()
        {
            PortFactory.MvcPort = 11625;
            this.server.StartServer(MvcApplication.Path, PortFactory.MvcPort, "/", "localhost");
        }


        [TestMethod]
        public void TestMvcPage()
        {
            MvcApprovals.VerifyMvcPage<TestableExampleController>(c => c.TestName, PathScrubbers.ScrubPath());
        }

        [TestCleanup]
        public void TearDown()
        {
            this.server.StopServer();
        }
    }

    public class TestableExampleController : TestableController<ExampleController>
    {
        public TestableExampleController(ExampleController t)
            : base(t)
        {
        }

        public ActionResult TestName()
        {
            DeleteEntireHardDrive();
            return ControllerUnderTest.SaveName(new Person { Name = "Henrik" });
        }

        private void DeleteEntireHardDrive()
        {
            // Just Kidding
        }
    }
}
