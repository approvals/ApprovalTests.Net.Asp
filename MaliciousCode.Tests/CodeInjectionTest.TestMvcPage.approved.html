The assembly:
...\bin\Debug\MaliciousCode.Tests.dll

Is not allowed.
If you would like to allow this assembly. 
Please add a filter to your Global.asax page like

UnitTestBootStrap.Register("MaliciousCode.Tests.dll");